janusgraph-driver/src/main/java/org/janusgraph/core/attribute/Text.java

```diff
@@ -236,16 +533,168 @@ public boolean evaluateRaw(String value, String term) {
         public boolean isValidCondition(Object condition) {
             return condition instanceof String && StringUtils.isNotBlank(condition.toString());
         }
+
+        @Override
+        public boolean hasNegation() {
+            return true;
+        }
+
+        @Override
+        public JanusGraphPredicate negate() {
+            return Text.NOT_CONTAINS_FUZZY;
+        }
+
         @Override
         public String toString() {
             return "textContainsFuzzy";
         }
 
+    },
+
+    /**
+     * Whether the text doesnt contain a token is at X Levenshtein of a token (case insensitive)
+     * with X=:
+     * - 0 for strings of one or two characters
+     * - 1 for strings of three, four or five characters
+     * - 2 for strings of more than five characters
+     */
+    NOT_CONTAINS_FUZZY {
+        @Override
+        public boolean test(Object value, Object condition) {
+            this.preevaluate(value, condition);
+            return value != null && evaluateRaw(value.toString(), (String) condition);
+        }
+
+        @Override
+        public boolean evaluateRaw(String value, String term) {
+            for (String token : tokenize(value.toLowerCase())) {
+                if (isFuzzy(term.toLowerCase(), token)) return false;
+            }
+            return true;
+        }
+
+        @Override
+        public boolean isValidCondition(Object condition) {
+            return condition instanceof String && StringUtils.isNotBlank(condition.toString());
+        }
+
+        @Override
+        public boolean hasNegation() {
+            return true;
+        }
+
+        @Override
+        public JanusGraphPredicate negate() {
+            return Text.CONTAINS_FUZZY;
+        }
+
+        @Override
+        public String toString() {
+            return "textNotContainsFuzzy";
+        }
+
+    },
+
+    /**
+     * Whether the text contains a given token sequence in the text (case insensitive)
+     */
+    CONTAINS_PHRASE {
+
+        @Override
+        public boolean test(Object value, Object condition) {
+            this.preevaluate(value, condition);
+            return value != null && evaluateRaw(value.toString(), (String) condition);
+        }
+
+        @Override
+        public boolean evaluateRaw(String value, String terms) {
+            List<String> valueTerms = tokenize(value.trim().toLowerCase());
+            List<String> tokenTerms = tokenize(terms.trim().toLowerCase());
+            if (!terms.isEmpty() && tokenTerms.isEmpty()) return false;
```

Comment:
Seems this if condition always evaluates to false because you have `StringUtils.isNotBlank((String) condition)` in `isValidCondition` method.

Reply:
Actually no, the tokenize method uses a MIN_TOKEN_LENGTH to ignore terms that are only one character long. So a short input like `a`, or one with several short terms like `a b c`, will fail out here but pass the isNotBlank check.

Reply:
Thanks, I didn't know that. Then I presume at least `!terms.isEmpty()` is always true, right?

Reply:
I assumed that too, this is another line that I copied over from an existing predicate (CONTAINS), and I wasnt sure if there was an edge case that I wasnt aware of.

Reply:
Fair enough.